File: MusicWeb.Server/MusicWeb.Models/Entities/Users/UserObservedArtist.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Entities.Posts;
using MusicWeb.Models.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class UserObservedArtist : BaseEntity
    {
        public DateTime ObservedDate { get; set; }

        public string UserId { get; set; }
        public int ArtistId { get; set; }
        public virtual Artist Artist{ get; set; }
        public virtual ApplicationUser User { get; set; }

        public virtual ICollection<Post> Posts { get; set; }
    }
}

File: MusicWeb.Client/src/models/UserObservedArtist.js
export default class UserObservedArtist {
  id = null;
  userId = null;
  artistId = null;
  observedDate = null;
  name = "";
  artist = {};
  user = {};
  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.userId = $data.userId;
      this.artistId = $data.artistId;
      this.name = $data.name;
      this.observedDate = $data.observedDate;
      this.artist = $data.artist ? $data.artist : {};
      this.user = $data.user ? $data.user : {};
    }
  }
}
